python/test/unit/language/test_core.py

```diff
@@ -5457,3 +5457,39 @@ def kernel(input):
 
     kernel[grid](input)
     assert torch.all(input == torch.tensor(grid, device=device))
+
+
+# -----------------------
+# test extern functions
+# -----------------------
+
+
+@pytest.mark.parametrize("dtype_str", ['float32', 'float64'])
+def test_math_extern(dtype_str):
+
+    @triton.jit
+    def kernel(
+        x_ptr,
+        y_ptr,
+        n_elements,
+        BLOCK_SIZE: tl.constexpr,
+    ):
+        pid = tl.program_id(axis=0)
+        block_start = pid * BLOCK_SIZE
+        offsets = block_start + tl.arange(0, BLOCK_SIZE)
+        mask = offsets < n_elements
+        x = tl.load(x_ptr + offsets, mask=mask)
+        y = GENERATE_TEST_HERE
+        tl.store(y_ptr + offsets, y, mask=mask)
+
+    shape = (128, )
+    rs = RandomState(17)
+    kernel = patch_kernel(kernel, {'GENERATE_TEST_HERE': 'libdevice.tanh(x)'})
```

Comment:
Does this need to be patched?  Seems like you can just write this line in the test directly?

Reply:
Sure, modified.